olive/workflows/run/config.py

```diff
@@ -136,6 +138,10 @@ def _resolve_config_str(v, values, alias, component_name="systems"):
     components = values[component_name] or {}
     if sub_component not in components:
         raise ValueError(f"{alias} {sub_component} not found in systems")
+    if component_name == "systems" and components[sub_component].type == "AzureML":
```

Comment:
can you move this auto insert to `_resolve_system` at line `149`? 

I think it can be something like 
```
def _resolve_system(v, values, system_alias, component_name="systems"):
    v = _resolve_config_str(v, values, system_alias, component_name=component_name) 
    v[system_alias] = validate_config(v[system_alias], SystemConfig)
    if v[system_alias].type == "AzureML":
         .... auto insert code .... 
``` 

Otherwise, it will only be auto inserted if the system is referenced by name. We also accept system config dicts directly.

Reply:
Sure will do. This default `component_name="systems"` confused me a lot. If there is no specific use case here, I'll remove this.

Reply:
Yes, it will always be "systems" for `_resolve_system`, so it is redundant. 

`_resolve_config_str` also has direct references to `system` which should actually be `component_name`. Looks like the PR that generalized the `resolve_config_str` code missed them.

Reply:
Yeah I mean the default value in `_resolve_config_str` for system. We also use it for data_config right now. I've removed all system keyword.

Reply:
Thanks! it looks good to me now